### PR TITLE
Fix windows exporter defaults not being set correctly

### LINKED ISSUE #4104: v0.34 Issue with windows_exporter incorrectly setting defaults
### What's wrong?

Defaults for some collectors (at least mssql) are not being applied correctly.

### Steps to reproduce

Add mssql as an enabled-collector but no specific mssql  sub-collectors. Normally the defaults would fill in several but currently it sets it to a blank string.

### System information

_No response_

### Software version

_No response_

### Configuration

```text
server:
  log_level: debug
metrics:
  global:
    scrape_interval: 60s
    remote_write:
      - url: https://prometheus-us-central1.grafana.net/api/prom/push 
        basic_auth: 
          username: 12345
          password: "PASSWORD"
integrations:
  agent:  
    enabled: true
  windows_exporter:
    enabled: true
    enabled_collectors: "mssql,logical_disk"
```

### Logs

_No response_

### PR DESCRIPTION
**Pull request by mattdurham**
#### PR Description

Windows_Exporter defaults were not being applied correctly and there was a race condition on reusing globals when modules + prometheus.exporter.windows. Forked the windows_exporter and will work on getting those changes moved.

#### Which issue(s) this PR fixes

Closed #4104

#### Notes to the Reviewer

The real bulk of the work happened in https://github.com/grafana/windows_exporter/pull/20, which I am going to work to get merged upstream.

#### PR Checklist

- [X] CHANGELOG updated
- [X] Documentation added
- [X] Tests updated

### CODE REVIEW

**Comment by mattdurham** [comment]
I looked over all the setting of flags, but would def like a double or triple check

**Comment by mattdurham** [comment]
@jkroepke would also love your input if able

File: pkg/integrations/windows_exporter/config_windows_test.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+func TestConfig(t *testing.T) {
+	built, total := testConfig(t, "")
+	// Default which is windows_exporter defaults minus textfile
+	require.Len(t, built, 7)
+	// Total should be 50
+	require.Len(t, total, 50)
+}
+
+func TestMultipleConfig(t *testing.T) {
```
Comment (jkroepke) [inline pkg/integrations/windows_exporter/config_windows_test.go:21]: I'm missing a test that `mssql.enabled_classes` is omit here, that the defaults are applied from upstream exporter. Something what the original bug report covered.

Comment (mattdurham) [reply pkg/integrations/windows_exporter/config_windows_test.go:21]: Good call, adding that test now.

**Review by erikbaranowski — APPROVED** [review]
LGTM

**Pull request merged**